packages/storage/test/iam.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var assert = require('assert');
+var extend = require('extend');
+var proxyquire = require('proxyquire');
+var util = require('@google-cloud/common').util;
+
+var promisified = false;
+var fakeUtil = extend({}, util, {
+  promisifyAll: function(Class) {
+    if (Class.name === 'Iam') {
+      promisified = true;
+    }
+  }
+});
+
+describe('storage/iam', function() {
+  var Iam;
+  var iam;
+
+  var BUCKET_INSTANCE;
+
+  before(function() {
+    Iam = proxyquire('../src/iam.js', {
+      '@google-cloud/common': {
+        util: fakeUtil
+      }
+    });
+  });
+
+  beforeEach(function() {
+    BUCKET_INSTANCE = {
+      id: 'bucket-id',
+      request: util.noop
+    };
+
+    iam = new Iam(BUCKET_INSTANCE);
+  });
+
+  describe('initialization', function() {
+    it('should promisify all the things', function() {
+      assert(promisified);
+    });
+
+    it('should localize the request function', function(done) {
```

Comment:
I had to take a second look at this one, because I find it a little confusing. Why not do something like this

```js
BUCKET_INSTANCE.request = function(callback) {
  assert.strictEqual(this, BUCKET_INSTANCE);
  callback(); // done fn
};

var iam = new Iam(BUCKET_INSTANCE);
iam.request_(done);
```